### ISSUE TITLE
snpe do not use encodings from aimet

### ISSUE THREAD

**Issue by mym2009** [quic/aimet#670]
I want to compare the difference between aimet simulation and inference on dsp, so I make a model only consist a convolution layer which weights are 1.0f and bias is 0. Then convert model.onnx to model.dlc by using snpe-onnx-dlc, and convert model.dlc to model_quantized.dlc by using aimet encodings. In dsp inference, through snpe inputMap.getUserBuffer(name)->getEncoding() api, I found the scale and offset changes with different input image and are not same as origin encodings from snpe-dlc-quantize. Do I miss some configuration about aimet or snpe?  Thanks.

**Comment by quic-akhobare** [comment]
Please check the --override_params option in https://developer.qualcomm.com/docs/snpe/tools.html#tools_snpe-onnx-to-dlc

**Comment by quic-ssiddego** [comment]
Linking related query #600

**Comment by mym2009** [comment]
@quic-akhobare Thanks for your reply. When converting dlc to quantized.dlc, I use this --overrid_params. There is no difference between aimet encodings and dlc encoding from snpe-dlc-info. I can confirm that when range of input get large, the scale of snpe encoding get large.

**Comment by mym2009** [comment]
@quic-akhobare I want to confirm that whether snpe will use all encodings(json format) and does not change in dsp run time. In my test, encodings from outputMap seems change with input image.

**Comment by quic-akhobare** [comment]
Hi @mym2009.. Sorry I messed up
In the converter look for an option like this --quantization_overrides <file> and then in snpe-dlc-quantize you specify --override_params

**Comment by quic-akhobare** [comment]
> @quic-akhobare I want to confirm that whether snpe will use all encodings(json format) and does not change in dsp run time. In my test, encodings from outputMap seems change with input image.

The idea is that all imported encodings will be preserved. Sometimes there are additional encodings needed because of differences between the graph IR from PyTorch to ONNX.. So there might be some additional ones

**Comment by mym2009** [comment]
@quic-akhobare Thanks for your reply. Before use aimet to retrain, I must know the difference aimet simulation and dsp inference. So I made a  model which only consist of one conv layer. Here is the detailed configuration:
1. pytorch input is 1x224x224x1, conv weights is 1.0, bias is 0.0 and nn.Conv2d(1,1,1), save onnx format and encoding
2. convert onnx to dlc with  --quantization_overrides, and override_params
3. use snpe examples NativeCpp SampleCode with USERBUFFER_FLOAT and dsp runtime.
This is the result( abs(sim-dsp)/dsp). It has big error between simulation and actual dsp inference. Do I have missed something or this is normal result? Looking forward to your favourable reply.
![conv](https://user-images.githubusercontent.com/68063596/125219730-eadbb980-e2f7-11eb-91c5-68c323c26f9f.png)

**Comment by quic-akhobare** [comment]
Hi @mym2009.. Good to know you were able to get the "pipeline" going..

I am not sure this is a great way of comparing sim vs. target. My suggestion would be to 

- try a more regular model, even if its something small - e.g. like mnist
- When comparing sim vs. target, base the comparisons on an accuracy metric rather than just looking at the diff in the raw output from the model.

**Comment by mym2009** [comment]
@quic-akhobare Thanks， I will try a regular model.
